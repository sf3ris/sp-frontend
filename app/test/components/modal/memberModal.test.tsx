import React from 'react';
import { shallow } from 'enzyme';
import MemberNewModalComponent from '../../../src/component/member/Modal/memberNewModal.component';

describe('Member Modal Component', () => {

    const onSaveFn = jest.fn();
    const onToggleFn = jest.fn();

    it('Should match snapshot', () => {

        const wrapper = shallow(
            <MemberNewModalComponent
                onSave={onSaveFn}
                isOpen={true}
                toggle={onToggleFn} />
        )

        expect(wrapper.debug()).toMatchSnapshot();

    });

    it('Should trigger save', () => {

        const wrapper = shallow(
            <MemberNewModalComponent
                onSave={onSaveFn}
                isOpen={true}
                toggle={onToggleFn} />
        )

        const saveButton = wrapper.find('#idSaveButton');
        
        expect(saveButton.length).toBe(1);
            
        saveButton.simulate('click');

        expect(onSaveFn).toBeCalled();

    });

    it('Should trigger toggle', () => {

        const wrapper = shallow(
            <MemberNewModalComponent
                onSave={onSaveFn}
                isOpen={true}
                toggle={onToggleFn} />
        )

        const discardButton = wrapper.find('#idDiscardButton');
        
        expect(discardButton.length).toBe(1);
            
        discardButton.simulate('click');

        expect(onToggleFn).toBeCalled();

    });
    
})
