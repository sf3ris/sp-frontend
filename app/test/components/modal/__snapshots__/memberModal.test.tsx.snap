// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Modal Component Should match snapshot 1`] = `
"<Fragment>
  <div className=\\"makeStyles-root-1\\">
    <div className=\\"makeStyles-textContainer-2\\">
      <MemberTextField label=\\"Cognome\\" value=\\"\\" onChange={[Function: onLastNameChange]} />
      <MemberTextField label=\\"Nome\\" value=\\"\\" onChange={[Function: onNameChange]} />
      <MuiPickersUtilsProvider utils={[Function: DateFnsUtils]}>
        <PickerWithState className=\\"makeStyles-selectField-3\\" margin=\\"normal\\" label=\\"Data di nascita\\" format=\\"dd/MM/yyyy\\" value={{...}} onChange={[Function: onChange]} minDate={{...}} maxDate={{...}} invalidDateMessage=\\"Invalid Date Format\\" minDateMessage=\\"Date should not be before minimal date\\" maxDateMessage=\\"Date should not be after maximal date\\" allowKeyboardControl={true} openTo=\\"date\\" views={{...}} />
      </MuiPickersUtilsProvider>
      <MemberTextField label=\\"Luogo di nascita\\" value=\\"\\" onChange={[Function: onBirthPlaceChange]} />
      <WithStyles(ForwardRef(InputLabel)) className=\\"makeStyles-selectField-3\\" id=\\"gender-select-id\\">
        Sesso
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select)) labelId=\\"gender-select-id\\" value=\\"\\" onChange={[Function: onGenderChange]} className=\\"makeStyles-selectField-3\\">
        <WithStyles(ForwardRef(MenuItem)) value=\\"M\\">
          M
        </WithStyles(ForwardRef(MenuItem))>
        <WithStyles(ForwardRef(MenuItem)) value=\\"F\\">
          F
        </WithStyles(ForwardRef(MenuItem))>
      </WithStyles(ForwardRef(Select))>
      <br />
      <MemberTextField label=\\"Codice Fiscale\\" value=\\"\\" maxLength={16} onChange={[Function: onFiscalCodeChange]} />
      <MemberTextField label=\\"Indirizzo\\" value=\\"\\" onChange={[Function: onAddressChange]} />
      <MemberTextField label=\\"CAP\\" maxLength={5} value=\\"\\" onChange={[Function: onZipCodeChange]} />
      <MemberTextField label=\\"CittÃ \\" value=\\"\\" onChange={[Function: onCityChange]} />
      <MemberTextField label=\\"Provincia\\" maxLength={2} value=\\"\\" onChange={[Function: onProvinceChange]} />
      <MemberTextField label=\\"Telefono\\" value=\\"\\" onChange={[Function: onPhoneChange]} />
      <MemberTextField label=\\"Email\\" value=\\"\\" onChange={[Function: onEmailChange]} />
    </div>
  </div>
  <div className=\\"makeStyles-textContainer-2\\">
    <WithStyles(ForwardRef(Button)) id=\\"idDiscardButton\\" variant=\\"contained\\" color=\\"secondary\\" onClick={[Function: onClose]}>
      Annulla
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button)) id=\\"idPopulatedButton\\" variant=\\"contained\\" color=\\"secondary\\" onClick={[Function: onPopulate]}>
      Popola
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button)) id=\\"idSaveButton\\" variant=\\"contained\\" color=\\"primary\\" onClick={[Function: onSave]} style={{...}}>
      Salva
    </WithStyles(ForwardRef(Button))>
  </div>
</Fragment>"
`;
